openqaoa/problems/problem.py

```diff
@@ -118,6 +118,10 @@ def __init__(self, n, terms, weights, clean_terms_and_weights=False):
         self.constant = constant
         self.n = n
 
+        self.name = "generic_qubo" if "name" not in metadata else metadata["name"]
```

Comment:
Is this attribute needed? this info is already in the metadata.

Or should be removed from the metadata?

Reply:
But, do we keep both `my_problem.name` and `my_problem.metadata.name`?

Reply:
The problem is that we are being inconsistent: why are we storing only some of the parameters and not others? And why are they being duplicated?

There are two type of data we care about when it comes to problems:
- the internal ones are all already stored as attributes of the object
- the external are the metadata tags that the users adds (e.g. the type of graph, the weighting strategy, etc)

I think that at the level of the object, we don't need to add the attributes to the metadata.

However, we need to make sure that when we get the input problem in the result of a workflow. There we need all the attributes (serialised) and the metadata. Whether we want things here to have a different structure (e.g. `n_cities` is stored in the metadata) can be discussed